tachyon/math/matrix/matrix_utils.h

```diff
@@ -74,6 +77,49 @@ std::vector<PackedPrimeField> PackRowVertically(
   });
 }
 
+// Expands a |RowMajorMatrix|'s rows from |rows| to |rows|^{|added_bits|},
+// moving values from row |i| to row |i|^{|added_bits|}. All new entries are set
+// to F::Zero()
+template <typename F>
+void ExpandWithZeroPad(RowMajorMatrix<F>& mat, size_t added_bits) {
+  if (added_bits == 0) {
+    return;
+  }
+
+  size_t original_rows = static_cast<size_t>(mat.rows());
+  size_t new_rows = static_cast<size_t>(mat.rows()) << added_bits;
+  size_t cols = static_cast<size_t>(mat.cols());
+
+  RowMajorMatrix<F> padded =
+      RowMajorMatrix<F>::Constant(new_rows, cols, F::Zero());
+
+  OPENMP_PARALLEL_FOR(size_t row = 0; row < original_rows; ++row) {
+    size_t padded_row_index = row << added_bits;
+    // TODO(ashjeong): Check if moved properly
+    padded.row(padded_row_index) = std::move(mat.row(row));
+  }
+  mat = std::move(padded);
+}
+
+// Swaps rows of a |RowMajorMatrix| such that each row is changed to the row
+// accessed with the reversed bits of the current index
+template <typename F>
+void ReverseMatrixIndexBits(RowMajorMatrix<F>& mat) {
+  size_t rows = static_cast<size_t>(mat.rows());
+  if (rows == 0) {
+    return;
+  }
+  CHECK(base::bits::IsPowerOfTwo(rows));
+  size_t log_n = base::bits::SafeLog2Ceiling(rows);
```

Comment:
I've changed instances of `log_n` to this-> log_size_of_group_` since I realized both are the same (along with a couple other variables as well, so log_n doesn't exist anymore. Though I do wanted to additionally ask what you meant by "Why not use CountTrailingZeroBits for consistency?" this!

Reply:
@ashjeong 
1. It seems like it's not yet applied for using `log_size_of_group`? Am I right?
2. I think I've seen you use `CountTrailingZeroBits` somewhere (but can't find at the moment) so I thought you could just use `CountTrailingZeroBits` here as well for consistency.

Reply:
Ah yep my bad - this instance is in `matrix_utils` so we don't use `this-> log_size_of_group_`.  As for `base::bits::SafeLog2Ceiling`, it was being used in `radix2_evaluation_domain` in cases where `CountTrailingZeroBits` would make more sense too (see [this comment](https://github.com/kroma-network/tachyon/pull/441#discussion_r1643715701)), so I followed the same convention here. Do you think we should change to `CountTrailingZeroBits` everywhere that it's needed?